### DISCUSSION TITLE
In Nextjs 13, how to fetch data in a server component once?

### DISCUSSION THREAD

**Discussion by mdegrees** [vercel/next.js#45453] [Help]
I'm building an app where I need the first render to be server-side rendered and the subsequent client-side transitions to be rendered using the client cache. I'm trying to find a way to prevent the server component from re-rendering and refetching. I thought of using the referer value from the `headers()`. If the referer is an internal URL, I'll skip server fetching. otherwise do fetch.

Is this the right way to go about it? Is there a better way I'm not aware of? 

Edit: My question is about Next13 RSC. Here is what I did to solve this issue in Next12: https://github.com/vercel/next.js/discussions/19611#discussioncomment-972107

thank you very much.

**Comment by guitheengineer** [comment]
The only problem I'm seeing there is refetching, if you just don't want to do one more request, you can use the data that was fetched from the server on getServerSideProps through props instead of fetching on the client. 

Other fetch libraries like react-query and apollo have caching policies that you can set to use the cached data from the server instead of fetching again on the client.

**Reply by mdegrees** [reply]
Actually I was referring to the new implementation of React Server Components in Nextjs 13. `getServerSideProps` is how I was solving it in Next12. It was easy to do something like: use server state on first load, use browser state on client side transitions. Now that I upgraded I couldn't find a way to replicate that.

**Reply by JerryLeeCS** [reply]
is there an update to this?
